internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -242,6 +245,107 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) setHTTPSIgnitionURL(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall, secretRef types.NamespacedName) error {
+
+	if clusterInstall.Spec.IgnitionEndpoint == nil {
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), hiveext.WorkerAgentMachinePool)
```

Comment:
Is this function used **only** for day2 worker or for any day2 ? If for any, then why does it have hardcoded `hiveext.WorkerAgentMachinePool`? If only for day2 worker, then maybe we can make it clear in the name of the function?

Reply:
you are right. I think we talked about it once but missed in review. We should MachineConfigPoolName and not hard code any value

Reply:
We cannot blindly use MachineConfigPoolName here for two reasons

- `MachineConfigPoolName` is very often empty because it's a field used for override and has no default value; if anything, `GetEffectiveRole()` is a good fit
- this code sets ignition endpoint for the whole ClusterInstall object, whereas MachineConfigPoolName or EffectiveRole are properties of a single Agent; there is no any contract saying that all the Agents for a specific ClusterInstall will contain hosts with only a single role